rs/consensus/utils/src/lib.rs

```diff
@@ -991,4 +1005,35 @@ mod tests {
             get_oldest_idkg_state_registry_version(&idkg_without_pre_sigs, &state)
         );
     }
+
+    #[test]
+    fn test_ignore_disqualified_ranks() {
+        ic_test_utilities::artifact_pool_config::with_test_pool_config(|pool_config| {
+            const SUBNET_SIZE: u64 = 10;
+            let Dependencies { mut pool, .. } = dependencies(pool_config, SUBNET_SIZE);
+
+            let height = Height::new(1);
+
+            // We fill the validated pool with blocks from every rank and
+            // incrementally disqualify the lowest qualified rank. Each time
+            // we assert that it's ignored by [`find_lowest_ranked_proposals`].
+            let f = get_faults_tolerated(SUBNET_SIZE as usize) as u64;
+            for i in 0..f + 1 {
```

Comment:
nit
```suggestion
            for i in 0..=f {
```

Reply:
I actually think the exclusive range is easier to understand, because `f + 1` has special meaning for consensus. I would prefer if we keep it this way. Would that work?

Reply:
sure, works for me